test/functional/listsinceblock.py

```diff
@@ -14,7 +14,12 @@ def __init__(self):
         self.setup_clean_chain = True
         self.num_nodes = 4
 
-    def run_test (self):
+    def run_test(self):
+        self.test_reorg()
```

Comment:
Now that you've split this into sub-tests, can you move the lines:

```python
self.nodes[2].generate(101)
self.sync_all()
```

into the `run_test()` function before calling `test_reorg()`. It's best if sub-tests have no shared state between them. You should be able to re-order or skip tests without breaking them, and generating the chain in `test_reorg()` breaks that assumption.

We want sub-tests to be as independent as possible so if they break it's easier to debug where the problem is.